### PR TITLE
completeness: add exponential backoff for downloading files

### PR DESCRIPTION
**Pull request by abhidg**
Also do not return null for countries in completeness.json
when download failed.

### CODE REVIEW

File: ingestion/monitoring/completeness.py [hunk 1]
```diff
@@ -53,14 +56,23 @@ def completeness_s3(s3_object, endpoint_url=None):
...
-            bucket.download_file(s3_object.key, tmp.name)
-        except Exception as e:
-            logging.error(f"Failed to download with exception {e}")
+        for tries in range(MAX_TRIES):
+            try:
+                bucket.download_file(s3_object.key, tmp.name)
+                success = True
+                break
+            except Exception:
+                logging.info(f"Failed to download {s3_object}, retrying in {2**tries} seconds ...")
```
Comment (iamleeg) [inline ingestion/monitoring/completeness.py:68]: could probably look at the exception to decide whether it's temporary (worth retrying) or not, but definitely should still log it.

**Review by iamleeg — CHANGES_REQUESTED** [review]

File: ingestion/monitoring/completeness.py (hunk 1 above)
Comment (jim-sheldon) [reply ingestion/monitoring/completeness.py:68]: What was the exception that led to adding `sleep`?

**Comment by jim-sheldon** [comment]
Might it be better to get this data from the database? It seems a bit more straightforward that way.

**Comment by abhidg** [comment]
> Better to get this data from the database? Seems more direct and simple that way.

DB would get from S3 anyway for the country export, and doing a .find() would start a collection scan, which I'd like to avoid.

**Comment by jim-sheldon** [comment]
> > Better to get this data from the database? Seems more direct and simple that way.
> 
> DB would get from S3 anyway for the country export, and doing a .find() would start a collection scan, which I'd like to avoid.

But S3 gets its data from the database. This seems like a good place to use scheduled triggers https://docs.mongodb.com/realm/triggers/scheduled-triggers/ ... or maybe making the completeness counting part of the data export.

**Review by iamleeg — APPROVED** [review]

**Pull request merged**